src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -943,19 +961,23 @@ class IMain(@BeanProperty val factory: ScriptEngineFactory, initialSettings: Set
        */
       val evalResult = Request.this.value match {
         case NoSymbol => ""
-        case sym      => "lazy val %s = %s".format(lineRep.resultName, originalPath(sym))
+        case sym      => "lazy val %s = { compute ; %s }".format(lineRep.resultName, originalPath(sym))
       }
       // first line evaluates object to make sure constructor is run
+      val initLine = settings.YreplWrap.value match {
+        case "app" => s"($fullAccessPath main null)"
+        case _     => fullAccessPath
+      }
       // initial "" so later code can uniformly be: + etc
```

Comment:
This is still unconditionally affected. I don't have a good feeling of who might be affected by this, so I'd either move to 2.12 or make all changes to wrapping dependent on -Yrepl-wrap. (Or convince ourselves that it wouldn't affect anyone.)